pkg/proxy/access/clusteraccess.go

```diff
@@ -0,0 +1,40 @@
+package access
+
+import (
+	"net/url"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// ClusterAccess holds information needed to access user namespaces in a member cluster for the specific user via impersonation
+type ClusterAccess struct { // nolint:revive
+	// APIURL is the Cluster API Endpoint for the namespace
+	apiURL url.URL
+	// impersonatorToken is a token of the impersonator's Service Account, usually the impersonator SA belongs to a member toolchaincluster
+	impersonatorToken string
+	// client is the kube client for the cluster.
+	client client.Client
+	// username is the id of the user to use for impersonation
+	username string
+}
+
+func NewClusterAccess(apiURL url.URL, client client.Client, impersonatorToken, username string) *ClusterAccess {
+	return &ClusterAccess{
+		apiURL:            apiURL,
+		client:            client,
+		impersonatorToken: impersonatorToken,
+		username:          username,
+	}
+}
+
+func (a *ClusterAccess) APIURL() url.URL {
+	return a.apiURL
+}
+
+func (a *ClusterAccess) ImpersonatorToken() string {
+	return a.impersonatorToken
+}
+
+func (a *ClusterAccess) Username() string {
+	return a.username
+}
```

Comment:
you could easily export the parameters and drop these "getter" functions.

Reply:
We could but then we'd also open them up to being modifiable outside of this package which I'm not sure we need/want? But we could change it if we want later